frontend/src/metabase/visualizations/components/ChartSettings/types.ts

```diff
@@ -1,15 +1,45 @@
+import type { ComponentType } from "react";
+
+import type {
+  VisualizationSettingDefinition,
+  WidgetComponentDef,
+} from "metabase/visualizations/types";
 import type Question from "metabase-lib/v1/Question";
 import type { Series, VisualizationSettings } from "metabase-types/api";
 
-// this type is not full, we need to extend it later
-export type Widget = {
-  id: string;
+export type Widget<
+  K extends keyof VisualizationSettings = keyof VisualizationSettings,
+  SettingsWidget extends ComponentType<any> = ComponentType<any>,
+> = {
+  id: K;
+  value: VisualizationSettings[K];
   section: string;
-  hidden?: boolean;
-  props: Record<string, unknown>;
-  title?: string;
-  widget: (() => JSX.Element | null) | undefined;
-};
+  title: string;
+  hidden: boolean;
+  marginBottom: string;
```

Comment:
I think this one should be optional, maybe some other props as well

Reply:
Agreed, I think all the boolean ones should be optional.